### PR TITLE
[16.0][MIG] account_comment_template: Migration to version 16.0

### PR DESCRIPTION
**Pull request by Yadier-Tecnativa**
@Tecnativa TT42208

### CODE REVIEW

**Review by CarlosRoca13 — APPROVED** [review]
LGTM :+1:

**Review by RodrigoBM — APPROVED** [review]

**Comment by RodrigoBM** [comment]
@pedrobaeza merge this pr?

File: account_comment_template/tests/test_account_move_report.py [hunk 1]
```diff
@@ -3,68 +3,55 @@
...
-            {
-                "name": "Test journal sale",
-                "code": "TST-JRNL-S",
-                "type": "sale",
-                "company_id": self.company.id,
-            }
-        )
-        move_form = self._create_invoice()
-        self.invoice = move_form.save()
+        move_form = cls._create_invoice(cls)
```
Comment (pedrobaeza) [inline account_comment_template/tests/test_account_move_report.py]: `AccountTestInvoicingCommon` has a helper method for creating an invoice. You can use it and save lines of code.

File: account_comment_template/tests/test_account_move_report.py [hunk 2]
```diff
@@ -3,68 +3,55 @@
...
-        self.partner = self.env["res.partner"].create({"name": "Partner Test"})
-        self.partner.base_comment_template_ids = [
-            (4, self.before_comment.id),
-            (4, self.after_comment.id),
+
+@tagged("post_install", "-at_install")
+class TestAccountInvoiceReport(AccountTestInvoicingCommon):
+    @classmethod
+    def setUpClass(cls, chart_template_ref=None):
+        super().setUpClass(chart_template_ref=chart_template_ref)
```
Comment (pedrobaeza) [inline account_comment_template/tests/test_account_move_report.py:15]: Let's do this always in tests (or almost always, except you are testing the chatter messages) for improving test times [1]:

```suggestion
        super().setUpClass(chart_template_ref=chart_template_ref)
        cls.env = cls.env(
            context=dict(
                cls.env.context,
                mail_create_nolog=True,
                mail_create_nosubscribe=True,
                mail_notrack=True,
                no_reset_password=True,
                tracking_disable=True,
            )
        )
```

[1] https://youtu.be/mks7haGqmKY?t=396

File: account_comment_template/tests/test_account_move_report.py [hunk 3]
```diff
@@ -3,68 +3,55 @@
...
+        cls.company = cls.env.ref("base.main_company")
+        cls.base_comment_model = cls.env["base.comment.template"]
+        cls.move_obj = cls.env.ref("account.model_account_move")
```
Comment (pedrobaeza) [inline account_comment_template/tests/test_account_move_report.py]: Incorrect variable name. It should be `move_model`. `move_obj` is intended for the handler of the model: `cls.env["account.move"]`.

File: account_comment_template/tests/test_account_move_report.py [hunk 4]
```diff
@@ -3,68 +3,55 @@
...
+        cls.before_comment = cls._create_comment(cls, position="before_lines")
+        cls.after_comment = cls._create_comment(cls, position="after_lines")
+        cls.partner = cls.env["res.partner"].create({"name": "Partner Test"})
+        cls.partner.base_comment_template_ids = [
```
Comment (pedrobaeza) [inline account_comment_template/tests/test_account_move_report.py]: Do this in the same `create` instruction.

File: account_comment_template/tests/test_account_move_report.py (hunk 2 above)
Comment (RodrigoBM) [reply account_comment_template/tests/test_account_move_report.py:15]: I like this

File: account_comment_template/tests/test_account_move_report.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
+# Copyright 2017 Simone Rubino - Agile Business Group
+# Copyright 2018 Tecnativa - Pedro M. Baeza
+# Copyright 2021-2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.tests import Form, tagged
```
Comment (RodrigoBM) [inline account_comment_template/tests/test_account_move_report.py]: Delete Form, its not used

File: account_comment_template/tests/test_account_move_report.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+            context=dict(
+                cls.env.context,
+                mail_create_nolog=True,
+                mail_create_nosubscribe=True,
+                mail_notrack=True,
+                no_reset_password=True,
+                tracking_disable=True,
+            )
+        )
+        cls.company = cls.env.ref("base.main_company")
```
Comment (RodrigoBM) [inline account_comment_template/tests/test_account_move_report.py]: Delete this line, its not used.

File: account_comment_template/tests/test_account_move_report.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        cls.base_comment_model = cls.env["base.comment.template"]
+        cls.move_obj = cls.env["account.move"]
```
Comment (RodrigoBM) [inline account_comment_template/tests/test_account_move_report.py]: you need to get the model id in odoo
```suggestion
        cls.res_model_id = cls.env.ref("account.model_account_move")
```

File: account_comment_template/tests/test_account_move_report.py [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+    def _create_comment(self, position):
+        return self.base_comment_model.create(
+            {
+                "name": "Comment " + position,
+                "company_id": self.company_data["company"].id,
+                "position": position,
+                "text": "Text " + position,
+                "models": "account.move",
+                "model_ids": [(6, 0, self.move_obj.ids)],
```
Comment (RodrigoBM) [inline account_comment_template/tests/test_account_move_report.py]:
```suggestion
                "model_ids": [(6, 0, self.res_model_id.ids)],
```

File: account_comment_template/tests/test_account_move_report.py [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+                mail_create_nosubscribe=True,
+                mail_notrack=True,
+                no_reset_password=True,
+                tracking_disable=True,
+            )
+        )
+        cls.company = cls.env.ref("base.main_company")
+        cls.base_comment_model = cls.env["base.comment.template"]
+        cls.move_obj = cls.env["account.move"]
+        cls.before_comment = cls._create_comment(cls, position="before_lines")
```
Comment (RodrigoBM) [inline account_comment_template/tests/test_account_move_report.py]:
```suggestion
        cls.before_comment = cls._create_comment(cls, "before_lines")
```

File: account_comment_template/tests/test_account_move_report.py [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+        cls.after_comment = cls._create_comment(cls, position="after_lines")
```
Comment (RodrigoBM) [inline account_comment_template/tests/test_account_move_report.py]:
```suggestion
        cls.after_comment = cls._create_comment(cls, "after_lines")
```

File: account_comment_template/tests/test_account_move_report.py [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+        cls.partner = cls.env["res.partner"].create({"name": "Partner Test"})
+        cls.partner.base_comment_template_ids = [
+            (4, cls.before_comment.id),
+            (4, cls.after_comment.id),
+        ]
+        cls.invoice = cls.init_invoice(
+            "out_invoice", products=cls.product_a + cls.product_b
```
Comment (RodrigoBM) [inline account_comment_template/tests/test_account_move_report.py]: you need to pass the partner
```suggestion
            "out_invoice", partner=cls.partner, products=cls.product_a + cls.product_b
```

File: account_comment_template/tests/test_account_move_report.py [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+    def test_comments_in_invoice_report(self):
+        res = self.env["ir.actions.report"]._render_qweb_html(
+            "account.report_invoice", self.invoice.ids
+        )
+        self.assertRegex(str(res[0]), self.before_comment.text)
+        self.assertRegex(str(res[0]), self.after_comment.text)
+
+    def test_comments_in_invoice(self):
+        new_invoice = self.init_invoice(
+            "out_invoice", products=self.product_a + self.product_b
```
Comment (RodrigoBM) [inline account_comment_template/tests/test_account_move_report.py]: you need to pass the partner
```suggestion
           "out_invoice",
            partner=self.partner,
            products=self.product_a + self.product_b,
```

**Review by RodrigoBM — CHANGES_REQUESTED** [review]

File: account_comment_template/tests/test_account_move_report.py (hunk 12 above)
Comment (Yadier-Tecnativa) [reply account_comment_template/tests/test_account_move_report.py]: Done

**Comment by Yadier-Tecnativa** [comment]
all suggestions resolved

**Review by RodrigoBM — APPROVED** [review]

**Comment by RodrigoBM** [comment]
its now ok @pedrobaeza ?

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**